### PR TITLE
#123 monitoring group page test

### LINKED ISSUE #123: TEST-123: Unit and UI tests for MonitoringGroupPage
Children of Issue[#97]

- [x] MonitoringGroupPageViewModel - unit test

- [x] MonitoringGroupPageActivity - UI test

### PR DESCRIPTION
**Pull request by KhangZhengNg**
### Related Issues
#123 

### Description
Update MonitoringGroupPageActivity and add relevant tests

### How to check PR
Run MonitoringGroupPageActivityTest / MonitoringGroupPageActivity

### CODE REVIEW

File: app/src/main/java/com/example/bigowlapp/activity/MonitoringGroupPageActivity.java [hunk 1]
```diff
@@ -92,11 +97,12 @@ private void setupSearchBar() {
         searchUsers.addTextChangedListener(new TextWatcher() {
             @Override
             public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {
+
             }
 
             @Override
             public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {
-                searchUsers(charSequence.toString().toLowerCase());
+                searchUsers(charSequence.toString().toLowerCase(Locale.getDefault()));
```
Comment (refatK) [inline app/src/main/java/com/example/bigowlapp/activity/MonitoringGroupPageActivity.java:103]: nice Lint fix

**Review by refatK — APPROVED** [review]
Looks good, and like the little cleanups in the code.

Good coverage increase!

File: app/src/main/java/com/example/bigowlapp/activity/MonitoringGroupPageActivity.java [hunk 2]
```diff
@@ -124,47 +128,60 @@ public boolean onContextItemSelected(MenuItem item) {
...
 
-        switch (menuItemName) {
-            case "View Profile":
-                break;
-            case "Remove": {
-                mGroupPageViewModel.removeUserFromGroup(contextMenuSelectedUser);
-                mUsers.remove(contextMenuSelectedUser);
-                resetUsersListViewAdapter();
-            }
+        if ("Remove".equals(menuItemName)) {
```
Comment (ShawnGuerra-Bautista) [inline app/src/main/java/com/example/bigowlapp/activity/MonitoringGroupPageActivity.java]: Switch the order. It should be menuItemName.equals("Remove");

File: app/src/main/java/com/example/bigowlapp/activity/MonitoringGroupPageActivity.java [hunk 3]
```diff
@@ -66,20 +68,21 @@ private void subscribeToData() {
...
-            // TODO: better error or allow view page when accessing group with no users
-            if (group == null || group.getMemberIdList() == null || group.getMemberIdList().isEmpty()) {
-                this.noGroupAlert().show();
-                return;
+            if (group == null) {
+                createAlertDialog("No monitoring group found",
+                        "Required to be the Monitor of a group before accessing this list");
+            } else if (group.getMemberIdList() == null || group.getMemberIdList().isEmpty()) {
+                createAlertDialog("No supervised member(s) found",
+                        "Required to have supervised member(s) before accessing this list");
```
Comment (ShawnGuerra-Bautista) [inline app/src/main/java/com/example/bigowlapp/activity/MonitoringGroupPageActivity.java:76]: Very good. You addressed the TODO.

**Review by ShawnGuerra-Bautista — CHANGES_REQUESTED** [review]
Very good.

Though, there are some changes to be made.

File: app/src/main/java/com/example/bigowlapp/activity/MonitoringGroupPageActivity.java (hunk 2 above)
Comment (KhangZhengNg) [reply app/src/main/java/com/example/bigowlapp/activity/MonitoringGroupPageActivity.java]: Thanks for pointing it out!

**Review by ShawnGuerra-Bautista — DISMISSED** [review]

**Review by ShawnGuerra-Bautista — APPROVED** [review]

**Pull request merged**